src/plugins/share/public/components/url_panel_content.tsx

```diff
@@ -404,17 +402,26 @@ export class UrlPanelContent extends Component<UrlPanelContentProps, State> {
   };
 
   private renderExportUrlAsOptions = () => {
+    const snapshotLabel = (
+      <FormattedMessage id="share.urlPanel.snapshotLabel" defaultMessage="Snapshot" />
+    );
     return [
       {
         id: ExportUrlAsType.EXPORT_URL_AS_SNAPSHOT,
-        label: this.renderWithIconTip(
-          <FormattedMessage id="share.urlPanel.snapshotLabel" defaultMessage="Snapshot" />,
-          <FormattedMessage
-            id="share.urlPanel.snapshotDescription"
-            defaultMessage="Snapshot URLs encode the current state of the {objectType} in the URL itself.
+        label: (
+          <>
+            {this.props.snapshotShareWarning
+              ? this.renderWithWarning(snapshotLabel, this.props.snapshotShareWarning)
+              : this.renderWithIconTip(
+                  snapshotLabel,
+                  <FormattedMessage
+                    id="share.urlPanel.snapshotDescription"
+                    defaultMessage="Snapshot URLs encode the current state of the {objectType} in the URL itself.
             Edits to the saved {objectType} won't be visible via this URL."
-            values={{ objectType: this.props.objectType }}
-          />
+                    values={{ objectType: this.props.objectType }}
+                  />
+                )}
+          </>
```

Comment:
This logic ensures that only a single icon tooltip is shown at one time, with the error/warning state taking precedence - i.e. if there is a snapshot share warning, then only the warning icon will be shown:

However, if there is no warning, then the info icon will be shown: